storage/aptosdb/src/db/include/aptosdb_reader.rs

```diff
@@ -943,7 +943,12 @@ impl AptosDB {
         let mut events_with_version = event_indices
             .into_iter()
             .map(|(seq, ver, idx)| {
-                let event = self.event_store.get_event_by_version_and_index(ver, idx)?;
+                let event = match self.event_store.get_event_by_version_and_index(ver, idx)? {
+                    ContractEvent::V2(_) => ContractEvent::V1(
+                        self.event_store.get_translated_v1_event_by_version_and_index(ver, idx)?,
```

Comment:
you should call the indexer db to get the translated data here?

Reply:
According to https://github.com/aptos-labs/aptos-core/blob/jpark%2Fevent-v2-translation/storage/aptosdb/src/event_store/mod.rs#L74-L74, it seems possible for `event_db` to access the indexer db? Am I missing something?

Reply:
I updated the code to use `self.state_store.internal_indexer_db`.

Reply:
```
    pub internal_indexer_db: Option<InternalIndexerDB>,
```
https://github.com/aptos-labs/aptos-core/blob/80c50219b2894713ab18433b19f5bad89e7e7511/storage/aptosdb/src/state_store/mod.rs#L110-L110

@grao1991 , by the way, do you know why it is optional?